src/layer.rs

```diff
@@ -1149,26 +1149,19 @@ where
         // uses the same reasoning as `parent_context` above.
         if let Some(follows_span) = ctx.span(follows) {
             let mut follows_extensions = follows_span.extensions_mut();
-            let _follows_data = follows_extensions
+            let follows_data = follows_extensions
                 .get_mut::<OtelData>()
                 .expect("Missing otel data span extensions");
-            // TODO:ban There are no tests that check this code :(
-            // TODO:ban if the follows span has a span builder the follows span should be _started_ here
-            // let follows_link = self.with_started_cx(follows_data, &|cx| {
-            //     otel::Link::with_context(cx.span().span_context().clone())
-            // });
-            // let follows_context = self
-            //     .tracer
-            //     .sampled_context(follows_data)
-            //     .span()
-            //     .span_context()
-            //     .clone();
-            // let follows_link = otel::Link::with_context(follows_context);
-            // if let Some(ref mut links) = data.builder.links {
-            //     links.push(follows_link);
-            // } else {
-            //     data.builder.links = Some(vec![follows_link]);
-            // }
+            let follows_context =
+                self.with_started_cx(follows_data, &|cx| cx.span().span_context().clone());
+            if let Some(builder) = data.builder.as_mut() {
+                builder
+                    .links
+                    .get_or_insert_with(Vec::new)
+                    .push(otel::Link::with_context(follows_context));
+            } else {
+                data.parent_cx.span().add_link(follows_context, vec![]);
```

Comment:
In the similar-ish `set_parent` case, we say that:

https://github.com/bantonsson/tracing-opentelemetry/blob/ba0d46e714db13291091af7282785a96dbc823c8/src/span_ext.rs#L178-L181

Because links don't enjoy the same "privileged" position as the parent, I don't think that constraint applies here ... right?

Reply:
Yes, there are no constraints like that, since a `Link` is just a special type of an `Attribute` with more bells and whistles.